cmd/cli/pipelines_init.go

```diff
@@ -0,0 +1,335 @@
+			Name: "generator",
+			Params: map[string]config.Parameter{
+				"format.type": {
+					Description: srcParams.Params["format.type"].Description,
+					Type:        srcParams.Params["format.type"].Type,
+					Default:     "structured",
+					Validations: srcParams.Params["format.type"].Validations,
+				},
+				"format.options.id": {
+					Description: "Generate field 'id' that's of the int type",
+					Type:        config.ParameterTypeString,
+					Default:     "int",
+				},
+				"format.options.name": {
+					Description: "Generate field 'name' that's of the string type",
+					Type:        config.ParameterTypeString,
+					Default:     "string",
+				},
+				"rate": {
+					Description: srcParams.Params["rate"].Description,
+					Type:        srcParams.Params["rate"].Type,
+					Default:     "1",
+				},
+			},
+		},
+		DestinationSpec: connectorTemplate{
+			Name: "log",
+		},
+	}
+}
+
+func (pi *PipelinesInit) getOutput() *os.File {
+	output, err := os.OpenFile(pi.configFilePath(), os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o600)
+	if err != nil {
+		log.Fatalf("error: failed to open %s: %v", pi.args.Path, err)
+	}
+
+	return output
+}
+
+func (pi *PipelinesInit) write(pipeline pipelineTemplate) error {
+	t, err := template.New("").Funcs(funcMap).Option("missingkey=zero").Parse(pipelineCfgTmpl)
+	if err != nil {
+		return cerrors.Errorf("failed parsing template: %w", err)
+	}
+
+	output := pi.getOutput()
+	defer output.Close()
+
+	err = t.Execute(output, pipeline)
+	if err != nil {
+		return cerrors.Errorf("failed executing template: %w", err)
+	}
+
+	return nil
+}
+
+func (pi *PipelinesInit) getSourceParams() (connectorTemplate, error) {
+	for _, conn := range builtin.DefaultBuiltinConnectors {
+		specs := conn.NewSpecification()
+		if specs.Name == pi.sourceConnector() || specs.Name == "builtin:"+pi.sourceConnector() {
+			if conn.NewSource == nil {
+				return connectorTemplate{}, cerrors.Errorf("plugin %v has no source", pi.sourceConnector())
+			}
+
+			return connectorTemplate{
+				Name:   specs.Name,
+				Params: conn.NewSource().Parameters(),
+			}, nil
+		}
+	}
+
+	return connectorTemplate{}, cerrors.Errorf("%v: %w", pi.sourceConnector(), plugin.ErrPluginNotFound)
+}
+
+func (pi *PipelinesInit) getDestinationParams() (connectorTemplate, error) {
+	for _, conn := range builtin.DefaultBuiltinConnectors {
+		specs := conn.NewSpecification()
+		if specs.Name == pi.destinationConnector() || specs.Name == "builtin:"+pi.destinationConnector() {
+			if conn.NewDestination == nil {
+				return connectorTemplate{}, cerrors.Errorf("plugin %v has no source", pi.destinationConnector())
+			}
+
+			return connectorTemplate{
+				Name:   specs.Name,
+				Params: conn.NewDestination().Parameters(),
+			}, nil
+		}
+	}
+
+	return connectorTemplate{}, cerrors.Errorf("%v: %w", pi.destinationConnector(), plugin.ErrPluginNotFound)
+}
+
+func (pi *PipelinesInit) configFilePath() string {
+	path := pi.args.Path
+	if path == "" {
+		path = "."
```

Comment:
Thinking a bit more about this, I believe that the default value should be `./pipelines`, because that's the default directory for pipelines, and also the directory created by `conduit init`. With this, getting started with Conduit would boil down to `conduit init && conduit pipelines init && conduit`